core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -125,27 +127,35 @@ oppia.factory('AnswerClassificationService', [
           return deferred.promise;
         }
 
-        if (result.outcome === defaultOutcome &&
-            INTERACTION_SPECS[oldState.interaction.id]
-              .is_string_classifier_trainable &&
+        var ruleBasedOutcomeIsDefault = (result.outcome === defaultOutcome);
+        var interactionIsTrainable = INTERACTION_SPECS[
+          oldState.interaction.id].is_interaction_trainable;
+        if (ruleBasedOutcomeIsDefault && interactionIsTrainable &&
             ENABLE_ML_CLASSIFIERS) {
-          var classifyUrl = '/explorehandler/classify/' + explorationId;
-          var params = (
-            isInEditorMode ? {} : LearnerParamsService.getAllParams());
-
-          $http.post(classifyUrl, {
-            old_state: oldState.toBackendDict(),
-            params: params,
-            answer: answer
-          }).then(function(response) {
-            var result = response.data;
+          var classifierData = StateClassifierMappingService.getClassifierData(
+            stateName);
+          var algorithmId = StateClassifierMappingService.getAlgorithmId(
+            stateName);
+          if (classifierData && algorithmId) {
+            var predictionService = (
+              PredictionAlgorithmRegistryService.getPredictionService(
+                algorithmId));
+            var predictedAnswerGroupIndex;
+            if (predictionService) {
+              predictedAnswerGroupIndex = predictionService.predict(
```

Comment:
This should be `predictedAnswerGroupIndex = predictionService.predict(classifier_data, answer);`. The predict function will return a singular value and not an object.